tests/integration/docarray_v2/test_issues.py

```diff
@@ -0,0 +1,41 @@
+from typing import List, Optional
+
+from docarray import BaseDoc, DocList
+
+from jina import Executor, Flow, requests
+
+
+class Nested2Doc(BaseDoc):
```

Comment:
can we copy the exact same structure as the issue?

Reply:
I think the only difference is the use of `Optional` and `List`. I added another case for this